src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2548,6 +2647,326 @@ public final void addBatch() throws SQLServerException {
         loggerExternal.exiting(getClassNameLogging(), "executeLargeBatch", updateCounts);
         return updateCounts;
     }
+    
+    
+    private String parseUserSQLForTableNameDW(boolean hasInsertBeenFound, boolean hasIntoBeenFound) {
+        // As far as finding the table name goes, There are two cases:
+        // Insert into <tableName> and Insert <tableName>
+        // And there could be in-line comments (with /* and */) in between.
+        // This method assumes the localUserSQL string starts with "insert".
+        localUserSQL = localUserSQL.trim();
+        if (checkAndRemoveComments()) {
+            return parseUserSQLForTableNameDW(hasInsertBeenFound, hasIntoBeenFound);
+        }
+        
+        if (localUserSQL.substring(0, 6).equalsIgnoreCase("insert") && !hasInsertBeenFound) {
+            localUserSQL = localUserSQL.substring(6);
+            return parseUserSQLForTableNameDW(true, hasIntoBeenFound);
+        }
+        
+        if (localUserSQL.substring(0, 4).equalsIgnoreCase("into") && !hasIntoBeenFound) {
+            // is it really "into"?
+            // if the "into" is followed by a blank space or /*, then yes.
+            if (localUserSQL.charAt(4) == ' ' || localUserSQL.charAt(4) == '\t' ||
```

Comment:
This is not accurate. The following query would fail.
```
insert into
table values(1)
```

Reply:
Thanks. This was a pretty important catch. I've made the change accordingly.